libs/datastructures/include/hpx/datastructures/variant_helper.hpp

```diff
@@ -0,0 +1,63 @@
+//  Copyright (c) 2020 Hartmut Kaiser
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef HPX_UTIL_VARIANT_HELPER_HPP
+#define HPX_UTIL_VARIANT_HELPER_HPP
+
+#include <hpx/config.hpp>
+#include <hpx/datastructures/config/defines.hpp>
+
+#if defined(HPX_HAVE_CXX17_STD_VARIANT)
+#include <hpx/type_support/pack.hpp>
+
+#include <cstddef>    // for size_t
+#include <utility>
+#include <variant>
+
+namespace hpx { namespace util {
+
+    // This is put into the same embedded namespace as the implementations in
+    // tuple.hpp
+    namespace adl_barrier {
+
+        template <std::size_t I, typename... Ts>
+        constexpr HPX_HOST_DEVICE HPX_FORCEINLINE typename
```

Comment:
```suggestion
        constexpr HPX_HOST_DEVICE HPX_FORCEINLINE
```
?

https://cdash.cscs.ch//viewBuildError.php?buildid=100368 and https://cdash.cscs.ch//viewBuildError.php?buildid=100419.

Reply:
MSVC is just a piece of work! Two `typename`'s in a row and not even a blink of an eye... Thanks!

Reply:
Well, in defense of MSVC, the older GCCs and clangs also didn't care ;)